### PR TITLE
fix(web): 404 error

### PR DESCRIPTION
**Pull request by martabal**
fix for #8590

## Screenshots

image

### CODE REVIEW

**Comment by martabal** [comment]
> While you're here, can you update the error page to also have a way back to the main timeline?

Maybe it's better to have a different error page for a 404 error with a "go back" button ?

**Comment by martabal** [comment]
Something like that ?

| Light | Dark |
| :---: | :---: |
| image | image |

**Comment by danieldietzler** [comment]
I personally like the the design but I'm not sure if it fits into the immich "design language"

**Comment by martabal** [comment]
> I personally like the the design but I'm not sure if it fits into the immich "design language"

Yeah, this is clearly not the immich "design language", I'm just wondering what's the best option between a full page error and the traditional error in a modal.

**Comment by bo0tzz** [comment]
For now I'd say stick with the existing error layout for simplicity, and it can get a more focused cleanup in the future. For now the error page is great to steer all the technical users in the right direction but when we head towards stable we might want to make it a bit more mom-friendly ;)

**Comment by jrasm91** [comment]
Personally, it we know it's a 404 error I think a full page thing would make sense.

File: web/src/hooks.client.ts [hunk 1]
```diff
@@ -3,10 +3,13 @@ import type { HandleClientError } from '@sveltejs/kit';
...
-const parseError = (error: unknown) => {
+const parseError = (error: unknown, statusCode: number) => {
   const httpError = isHttpError(error) ? error : undefined;
-  const statusCode = httpError?.status || httpError?.data?.statusCode || 500;
-  const message = httpError?.data?.message || (httpError?.data && String(httpError.data)) || httpError?.message;
+
+  const message =
+    statusCode === 404
+      ? 'This page does not exist'
+      : undefined || httpError?.data?.message || (httpError?.data && String(httpError.data)) || httpError?.message;
```
Comment (danieldietzler) [inline web/src/hooks.client.ts:12]: I would prefer splitting this up into two variables like it was before. This is quite long right now

**Review by danieldietzler — APPROVED** [review]

File: web/src/hooks.client.ts (hunk 1 above)
Comment (martabal) [reply web/src/hooks.client.ts:12]: Is it even possible to have different error messages with the current implementation ?

It's maybe easier to set ourselves different error messages with a switch case

Comment (jrasm91) [reply web/src/hooks.client.ts:12]: If the api server is down or unreachable it shows a different error.

File: web/src/routes/+error.svelte [hunk 2]
```diff
@@ -54,6 +55,11 @@
                 <label for="stacktrace">Stacktrace</label>
                 <pre id="stacktrace" class="text-xs">{$page.error?.stack || 'No stack'}</pre>
               {/if}
+              {#if $page.error?.code}
+                <a href="/">
+                  <Button size="sm">Go back</Button>
+                </a>
+              {/if}
```
Comment (jrasm91) [inline web/src/routes/+error.svelte:62]: should we not show this button all the time? I don't think it needs to be limited to when there is an error code.

Comment (martabal) [reply web/src/routes/+error.svelte:62]: I think we should keep it only for 404 errors, the button is at best useless for other errors at worst misleading

**Comment by martabal** [comment]
Closing in favor of #8901

**Pull request closed without merging**